### PR TITLE
Refresh: Upgrade to Requests 2.0.0

### PR DESCRIPTION
**Pull request by costdev**
### Upgrade references to the Requests library in the WP code

This PR:
- [x] Moves all new files to `src/wp-includes/Requests/src/`.
- [x] Updates the `require` statement for `Autoload.php` in `src/wp-includes/class-wp-http.php`.
- [x] Adds Requests 1.x files/directories to `$_old_files`.
- [x] Introduces `_preload_old_requests_files()` to preload Requests 1.x slated for deletion.
- [x] Adds a call to `_preload_old_requests_files()` at the beginning of `update_core()`.

Trac ticket: https://core.trac.wordpress.org/ticket/54504

Refresh of https://github.com/WordPress/wordpress-develop/pull/1624.

## How to Test?

Testing instructions are found here https://gist.github.com/hellofromtonya/bb0aa2d4b2311c8b40fa5ae8ada4dc19.

### CODE REVIEW

File: src/wp-admin/includes/update-core.php [hunk 1]
```diff
@@ -1478,6 +1547,36 @@ function update_core( $from, $to ) {
...
+/**
+ * Preloads old Requests files stored in $_old_files
+ * that may be needed during the update process.
+ *
+ * @since 6.2.0
+ *
+ * @param string $to Path to old WordPress installation.
+ */
+function _preload_old_requests_files( $to ) {
+	global $_old_files, $wp_filesystem;
```
Comment (mukeshpanchal27) [inline src/wp-admin/includes/update-core.php]: @costdev `global` documentation missing.

Comment (costdev) [reply src/wp-admin/includes/update-core.php]: Thanks @mukeshpanchal27! Fixed in [a375f6c](https://github.com/WordPress/wordpress-develop/pull/3732/commits/a375f6c107f2327b054bc7f121a1f5d28a9fcd38).

File: src/wp-includes/class-requests.php [hunk 2]
```diff
@@ -7,7 +7,30 @@
...
+		E_USER_DEPRECATED
+	);
+
+	// Prevent the deprecation notice from being thrown twice.
+	if (!defined('REQUESTS_SILENCE_PSR0_DEPRECATIONS')) {
+		define('REQUESTS_SILENCE_PSR0_DEPRECATIONS', true);
+	}
+}
+
+require_once __DIR__ . '/Requests/Requests.php';
```
Comment (hellofromtonya) [inline src/wp-includes/class-requests.php]:
```suggestion
require_once __DIR__ . '/Requests/src/Requests.php';
```

File: src/wp-includes/class-requests.php [hunk 3]
```diff
@@ -17,985 +40,39 @@
...
 	 * @codeCoverageIgnore
 	 *
 	 * @param string $class Class name to load
 	 */
 	public static function autoloader($class) {
-		// Check that the class starts with "Requests"
-		if (strpos($class, 'Requests') !== 0) {
-			return;
+		if (class_exists('WpOrg\Requests\Autoload') === false) {
+			require_once __DIR__ . '/Requests/Autoload.php';
```
Comment (hellofromtonya) [inline src/wp-includes/class-requests.php]:
```suggestion
			require_once __DIR__ . '/Requests/src/Autoload.php';
```

File: src/wp-includes/class-requests.php [hunk 4]
```diff
@@ -17,985 +40,39 @@
...
-			$parts[0] = '*';
-			$wildcard = implode('.', $parts);
-			if ($wildcard === $reference) {
-				return true;
-			}
-		}
-
-		return false;
+	public static function register_autoloader() {
+		require_once __DIR__ . '/Requests/Autoload.php';
```
Comment (hellofromtonya) [inline src/wp-includes/class-requests.php]:
```suggestion
		require_once __DIR__ . '/Requests/src/Autoload.php';
```

File: src/wp-includes/class-requests.php [hunk 5]
```diff
@@ -7,7 +7,30 @@
...
+		'The PSR-0 `Requests_...` class names in the Request library are deprecated.'
+		. ' Switch to the PSR-4 `WpOrg\Requests\...` class names at your earliest convenience.',
+		E_USER_DEPRECATED
+	);
+
+	// Prevent the deprecation notice from being thrown twice.
+	if (!defined('REQUESTS_SILENCE_PSR0_DEPRECATIONS')) {
+		define('REQUESTS_SILENCE_PSR0_DEPRECATIONS', true);
+	}
+}
```
Comment (swissspidy) [inline src/wp-includes/class-requests.php:31]: Lots of missing whitespace here between parentheses etc. Is this intentional / coming from Requests?

Comment (costdev) [reply src/wp-includes/class-requests.php:31]: Yeah it's coming from Requests and this file is excluded from `phpcs.xml.dist` - [See here](https://github.com/WordPress/wordpress-develop/blob/trunk/phpcs.xml.dist#L116).

**Review by hellofromtonya — APPROVED** [review]
My testing also confirms it works as expected. It resolves the blocking issues of filesystem case-insensitivity and fatal errors from the intermixing of Requests 1.x with 2.x.

This PR is ready for commit ✅

**Comment by hellofromtonya** [comment]
Commit via changeset https://core.trac.wordpress.org/changeset/54997.

**Pull request closed without merging**